src/components/Menu/MenuItem.js

```diff
@@ -61,9 +71,22 @@ const MenuDropdownListItem = (props) => {
   // if selected true or false, it's a menuitemcheckbox, else it's a menuitem
 
   return (
-    <li {...rest} className={cx(sldsClasses)} role="presentation">
-      <a role={role} onClick={onClick} aria-checked={selected}>
-        <span className="slds-truncate" title={children}>
+    <li
+      className={cx(sldsClasses)}
+      role="presentation"
+      onClick={onClick}
+      {...rest}
+    >
+      <a
+        aria-checked={selected}
+        role={role}
+        data-value={dataValue}
```

Comment:
I don't get why this is added here and not on the li since you access dataset on li in ControlledMenu.

Reply:
It does not work with `data-value` on the `li` for some reason. Then `data-value` is always undefined in the event.

Reply:
Hm that is odd. I'm gonna take a look